### DISCUSSION TITLE
Importing libvips in QT in windows

### DISCUSSION THREAD

**Discussion by ShanMing88** [libvips/libvips#3021] [Help wanted]
When importing libvips in qt in widows, it shows me Link error

![image](https://user-images.githubusercontent.com/112085684/188247804-2ea937ca-e7e7-4d37-829a-bf330ac67411.png)

Please help me to fix this issue urgently
Thanks

**Comment by jcupitt** [comment]
Hi @ShanMing88,

Your problem is with C++. It does not define an ABI, so all parts of a C++ project must be built with the same version of the same compiler.

I would take a copy of the libvips C++ binding from here:

https://github.com/libvips/libvips/tree/master/cplusplus

And copy-paste it into your project. Now when you build, all the interfaces will be MSVC, so it'll link correctly.

Your alternative would be to use the libvips C interface. C does define an ABI, so you can mix compilers.

**Reply by ShanMing88** [reply]
Hi @jcupitt ,
Thanks for your reply.
I downloaded the release() for Windows version here, https://github.com/libvips/libvips/releases/download/v8.8.0/vips-dev-w64-all-8.8.0.zip
And unzip all to my project root folder.
Then include the include folder and link the lib in po file of my project.
![Screenshot_3](https://user-images.githubusercontent.com/112085684/188301974-03db12a4-6006-460d-a333-86aaf9de3d85.png)
But I still got the link error when build the project.
![Screenshot_4](https://user-images.githubusercontent.com/112085684/188301988-ff827cb3-fe7c-45c9-ad32-b536a03178d1.png)
I think this error happened when call VImage::option() function.
Would you let me know what is wrong.
Regards,

**Reply by jcupitt** [reply]
Like I said, you can't mix C++ compilers. C++ does not support this. libvips is compiled with g++, and you are trying to link to it with MSVC.

You need to compile the libvips C++ layer with your C++ compiler.  It should be easy -- just copy those few files I linked into your project, then link your project against the C (not C++) DLLs.

**Reply by ShanMing88** [reply]
Hi, Jcupitt.
Would you guide me kindly how to compile the libvips C++ layer?

**Reply by jcupitt** [reply]
I've not actually done it myself, but I'm told it's straightforward. Just copy those six `.cpp` files into your project, plus the `.h` files in include, and build your project against the libvips C API.

**Comment by zjian361** [comment]
> When importing libvips in qt in widows, it shows me Link error
> 
> ![image](https://user-images.githubusercontent.com/112085684/188247804-2ea937ca-e7e7-4d37-829a-bf330ac67411.png)
> 
> Please help me to fix this issue urgently Thanks

Hi,ShanMing88,do you compile the libvips C++ layer successful?
